File: frontend/types/course.ts
export type SkillStatus = 'LOCKED' | 'NOT_STARTED' | 'IN_PROGRESS' | 'COMPLETED';

export interface SkillNodeData {
  id: string;
  title: string;
  description: string;
  status: SkillStatus;
  progress: number; // 0 to 100
  totalLessons: number;
  completedLessons: number;
  iconName?: string;
  positionOffset?: number; // X-offset for winding path (-50 to 50)
}

export interface UnitData {
  id: string;
  unitNumber: number;
  sectionNumber: number;
  title: string;
  description: string;
  bannerColor?: string;
  skills: SkillNodeData[];
}

export interface CourseData {
  id: string;
  title: string;
  language: string;
  flagEmoji: string;
  units: UnitData[];
}

File: backend/app/schemas/course.py
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel, ConfigDict
from app.schemas.lesson import LessonResponse

class SkillPathResponse(BaseModel):
    id: int
    unit_id: int
    title: str
    description: Optional[str] = None
    icon: Optional[str] = None
    order_index: int
    status: str  # LOCKED, NOT_STARTED, IN_PROGRESS, COMPLETED
    total_lessons: int
    completed_lessons: int
    lessons: List[LessonResponse] = []

    model_config = ConfigDict(from_attributes=True)

class UnitPathResponse(BaseModel):
    id: int
    course_id: int
    title: str
    description: Optional[str] = None
    order_index: int
    skills: List[SkillPathResponse] = []

    model_config = ConfigDict(from_attributes=True)

class CoursePathResponse(BaseModel):
    id: int
    name: str
    language: str
    description: Optional[str] = None
    image_url: Optional[str] = None
    units: List[UnitPathResponse] = []

    model_config = ConfigDict(from_attributes=True)
